src/providers/aws/mod.rs

```diff
@@ -66,6 +68,23 @@ impl AwsProvider {
         format!("{}/{}", URL, key)
     }
 
+    fn fetch_imdsv2_token(&self) -> Result<String> {
+        let tokendata: Option<String> = self
+            .client
+            .clone()
+            .header(header::HeaderName::from_bytes(b"X-aws-ec2-metadata-token-ttl-seconds")
+                        .chain_err(|| "setting header name for aws imdsv2 token")?,
+                    header::HeaderValue::from_bytes(b"21600")
+                        .chain_err(|| "setting header value for aws imdsv2 token")?)
+            .put(
+                retry::Raw,
+                format!("{}/{}", "http://169.254.169.254/latest", "api/token"),
```

Comment:
Why are we hardcoding `latest` here?

Reply:
Sorry I forgot to add a note on this.

So I've tried to fetch v2 token with different versions for example, 2009-04-04 and 2019-10-01, it seems like only `latest` is accepted and returned with a valid token, and others returned a 403 Forbidden.